ImageFunctions/Thumbnail.cs

```diff
@@ -0,0 +1,117 @@
+using System;
+using System.IO;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace ImageFunctions
+{
+    public static class Thunbnail
+    {
+        private static readonly string BLOB_STORAGE_CONNECTION_STRING = Environment.GetEnvironmentVariable("AzureWebJobsStorage");
+
+        private static string GetBlobNameFromUrl(string bloblUrl)
+        {
+            var uri = new Uri(bloblUrl);
+            var cloudBlob = new CloudBlob(uri);
+            return cloudBlob.Name;
+        }
+
+        private static IImageEncoder GetEncoder(string extension)
+        {
+            IImageEncoder encoder = null;
+
+            extension = extension.Replace(".", "");
+
+            var isSupported = Regex.IsMatch(extension, "gif|png|jpe?g", RegexOptions.IgnoreCase);
+
+            if (isSupported)
+            {
+                switch (extension)
+                {
+                    case "png":
+                        encoder = new PngEncoder();
+                        break;
+                    case "jpg":
+                        encoder = new JpegEncoder();
+                        break;
+                    case "jpeg":
+                        encoder = new JpegEncoder();
+                        break;
+                    case "gif":
+                        encoder = new GifEncoder();
+                        break;
+                    default:
+                        break;
+                }
+            }
+
+            return encoder;
+        }
+
+        [FunctionName("Thumbnail")]
+        public static async Task Run(
+            [EventGridTrigger]EventGridEvent eventGridEvent,
+            [Blob("{data.url}", FileAccess.Read)] Stream input,
+            ILogger log)
+        {
+            try
+            {
+                if (input != null)
+                {
+                    var createdEvent = ((JObject)eventGridEvent.Data).ToObject<StorageBlobCreatedEventData>();
+                    var extension = Path.GetExtension(createdEvent.Url);
+                    var encoder = GetEncoder(extension);
+
+                    if (encoder != null)
+                    {
+                        var thumbContainerName = Environment.GetEnvironmentVariable("THUMBNAIL_CONTAINER_NAME");
+                        var storageAccount = CloudStorageAccount.Parse(BLOB_STORAGE_CONNECTION_STRING);
+                        var blobClient = storageAccount.CreateCloudBlobClient();
+                        var container = blobClient.GetContainerReference(thumbContainerName);
+                        var blobName = GetBlobNameFromUrl(createdEvent.Url);
+                        var blockBlob = container.GetBlockBlobReference(blobName);
+
+                        using (var output = new MemoryStream())
+                        using (Image<Rgba32> image = Image.Load(input))
+                        {
+                            image.Mutate(x => x.Resize(image.Width / 2, image.Height / 2));
```

Comment:
Good catch. I had that logic in from a test and forgot to remove it. I have updated the logic to calculate the height based on a given width. I moved the thumbnail width out do an Environment variable and have defaulted it to 100 - so that should do the trick ;)